src/compiler/factory/emitHelpers.ts

```diff
@@ -961,6 +967,17 @@ namespace ts {
             };`
     };
 
+    export const classPrivateFieldInHelper: UnscopedEmitHelper = {
+        name: "typescript:classPrivateFieldIn",
+        importName: "__classPrivateFieldIn",
+        scoped: false,
+        text: `
+            var __classPrivateFieldIn = (this && this.__classPrivateFieldIn) || function(receiver, state) {
+                if (receiver === null || (typeof receiver !== "object" && typeof receiver !== "function")) throw new TypeError("Cannot use 'in' operator on non-object");
+                return typeof state === "function" ? receiver === state : state.has(receiver);

```

Comment:
I was only expecting `state.has(receiver)` -- when would `state` be a function and why does `receiver === state` mean that `#state in receiver` is true?

Reply:
This is for static fields. Where instead of using a `WeakMap` the receiver instead must be the class constructor.

Reply:
[`classPrivateFieldGetHelper`](https://github.com/microsoft/TypeScript/blob/c620266c87b52ed2c011b0ba8b599177ba934259/src/compiler/factory/emitHelpers.ts#L840-L887) and [`classPrivateFieldSetHelper `](https://github.com/microsoft/TypeScript/blob/c620266c87b52ed2c011b0ba8b599177ba934259/src/compiler/factory/emitHelpers.ts#L900-L950) have pretty comprehensive documentation on this. Maybe we can reference those in `classPrivateFieldInHelper` docs.

Reply:
If it's obvious to people familiar with the #private spec, it can be a pretty short link. It's just that I had never looked at the emit for static #privates.

Reply:
Doc-text added to helper now.